### DISCUSSION TITLE
unwanted but valid keyword in aucocomplete popup

### DISCUSSION THREAD

**Discussion by Jerther** [dbeaver/dbeaver#15879] [Ideas]
**Is your feature request related to a problem? Please describe.**
I use dbeaver for a postgresql database, and I use autocomplete. The keywords dbeaver suggests are generally useful, but there is a "notnull" (no space) keyworkd that keeps popping up and is selected by default when I write "where x is not null" and while I can just select "null" or write the whole text manually or press space, I find it very annoying.

**Describe the solution you'd like**
I'd like some way of removing that keyword form the autocomplete list, either via a GUI or a workaround.

**Describe alternatives you've considered**
None. I couldn't find anything about this anywhere, so I thought it was hardcoded or something, and I just live with it :(

**Additional context**
![image](https://user-images.githubusercontent.com/5856745/158628794-ebb35ef1-3700-4516-98f6-58300a22335f.png)

**Comment by LonwoLonwo** [comment]
Hello @Jerther 

As I can see in [PostgreSQL documentation](https://www.postgresql.org/docs/current/functions-comparison.html) - `NOTNULL` is a valid keyword.

![2022-03-16 20_27_04-DBeaver Enterprise 21 3 3 - _postgres_ Script-1](https://user-images.githubusercontent.com/45152336/158651236-4da3cb28-b4c7-423f-b027-c1814fbd5597.png)

**Comment by Jerther** [comment]
Hmmm, you're right... So the syntax is

`where x notnull`

Even more annoying ;)

Is there a way to remove or hide that keyword from the suggestion popup, even though it's valid?

**Comment by LonwoLonwo** [comment]
No, sorry. The keywords returned the driver DBeaver shows in the pop-up list. There is no specific keywords list in DBeaver with the word 'NOTNULL'.

**Comment by Jerther** [comment]
So it comes from the driver. I took a quick look around the [JDBC documentation for postgres](https://jdbc.postgresql.org/documentation/head/) and I couldn't find anything related to keywords. Well anyway, if dbeaver does not provide some kind of "blacklist" or "whitelist" for SQL keywords, I might as well give up on this.

**Comment by LonwoLonwo** [comment]
DBeaver has this kind of list of extra keywords, but 'NOTNULL' is not in this list.
https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDialect.java

**Reply by Jerther** [reply]
I think the "issue" I'm having would be alleviated by https://github.com/dbeaver/dbeaver/issues/6528#issuecomment-522035348

**Comment by timw2** [comment]
@LonwoLonwo how does DBeaver determine what order the autocomplete words are listed in? I think the frustration is that the word I typed is in the list but isn't the first option in the list that is selected so if I just type the query normally without thinking about autocomplete it gets "broken".

Could the autocomplete list put any options that exactly match what is typed (case insensitive) at the top of the list?
